modules/meteroid/crates/meteroid-store/src/repositories/schedules.rs

```diff
@@ -0,0 +1,94 @@
+use crate::errors::StoreError;
+use crate::{domain, Store, StoreResult};
+use error_stack::Report;
+use uuid::Uuid;
+
+#[async_trait::async_trait]
+pub trait ScheduleInterface {
+    async fn delete_schedule(&self, id: Uuid, auth_tenant_id: Uuid) -> StoreResult<()>;
+    async fn list_schedules(
+        &self,
+        plan_version_id: Uuid,
+        tenant_id: Uuid,
+    ) -> StoreResult<Vec<domain::Schedule>>;
+
+    async fn insert_schedule(
+        &self,
+        schedule: domain::ScheduleNew,
+        auth_tenant_id: Uuid,
+    ) -> StoreResult<domain::Schedule>;
+
+    async fn patch_schedule(
+        &self,
+        schedule: domain::SchedulePatch,
+        auth_tenant_id: Uuid,
+    ) -> StoreResult<domain::Schedule>;
+}
+
+#[async_trait::async_trait]
+impl ScheduleInterface for Store {
+    async fn delete_schedule(&self, id: Uuid, auth_tenant_id: Uuid) -> StoreResult<()> {
+        let mut conn = self.get_conn().await?;
+
+        diesel_models::schedules::Schedule::delete(&mut conn, id, auth_tenant_id)
+            .await
+            .map_err(Into::<Report<StoreError>>::into)
+            .map(|_| ())
+    }
+
+    async fn list_schedules(
+        &self,
+        plan_version_id: Uuid,
+        auth_tenant_id: Uuid,
+    ) -> StoreResult<Vec<domain::Schedule>> {
+        let mut conn = self.get_conn().await?;
+
+        diesel_models::schedules::Schedule::list(&mut conn, plan_version_id, auth_tenant_id)
+            .await
+            .map_err(Into::<Report<StoreError>>::into)?
+            .into_iter()
+            .map(TryInto::try_into)
+            .collect::<Result<Vec<_>, _>>()
+    }
+
+    async fn insert_schedule(
+        &self,
+        schedule: domain::ScheduleNew,
+        auth_tenant_id: Uuid,
+    ) -> StoreResult<domain::Schedule> {
+        let mut conn = self.get_conn().await?;
+
+        let insertable: diesel_models::schedules::ScheduleNew = schedule.try_into()?;
+
+        // make sure the plan version exists and belongs to auth tenant
+        diesel_models::plan_versions::PlanVersion::find_by_id_and_tenant_id(
+            &mut conn,
+            insertable.plan_version_id,
+            auth_tenant_id,
+        )
+        .await
+        .map_err(Into::<Report<StoreError>>::into)?;
+
+        insertable
+            .insert(&mut conn)
+            .await
+            .map_err(Into::<Report<StoreError>>::into)
+            .and_then(TryInto::try_into)
```

Comment:
this is one query in cornucopia, i struggled to make it in diesel as didn't want to make it as raw SQL so decided to have 2 queries in store instead, for this particular operation it should be fine.